File: docs/tutorial/magics/msl_uv200/solution/msl_uv200.f
      program  magics

C Defines the lines for the title
      character*25 title
      dimension title(2)

      data title /
     +    'My first line', 
     +    'My second line' /

C open Magics
      call popen

C setup the output name
      call psetc('output_name', 'msl_uv_australia')

C Setup the geographical area : here we use the Australia definition
      call psetr('subpage_lower_left_longitude', 80.000000)
      call psetr('subpage_upper_right_longitude', 190.000000)
      call psetr('subpage_upper_right_latitude', -5.000000)
      call psetc('subpage_map_projection', 'cylindrical')
      call psetr('subpage_lower_left_latitude', -55.000000)

C Setup the backgrounf coastlines
      call psetc('map_coastline_sea_shade', 'on')
      call psetc('map_coastline_sea_shade_colour', 'white')
      call psetc('map_label', 'off')
      call psetc('map_coastline_land_shade_colour', 'cream')
      call psetc('map_grid', 'off')
      call psetc('map_coastline_colour', 'tan')
      call psetc('map_coastline_land_shade', 'on')
      call pcoast

C Import the wind speed data
      call psetc('grib_input_file_name', '../speed200.grb')
      call pgrib

C Define the shading for the wind speed
      call psetc('legend', 'on')
      call psetc('contour_level_selection_type', 'level_list')
      call psetc('contour_shade_method', 'area_fill')
      call psetc('contour_shade', 'on')
      call pset1r('contour_level_list', 
     +           (/30., 40., 50., 60., 70., 80., 90., 100./), 8)
      call psetr('contour_hi_min_value', 15.000000)
      call psetr('contour_reference_level', 0.000000)
      call psetc('contour_hilo', 'on')
      call psetc('contour_hilo_format', '(F3.0)')
      call psetr('contour_hilo_height', 0.200000)
      call psetc('contour_highlight', 'off')
      call psetc('contour_label', 'off')
      call psetc('contour_hilo_type', 'number')
      call psetr('contour_hilo_suppress_radius', 30.000000)
      call psetc('contour_shade_max_level_colour', 'evergreen')
      call psetc('contour_shade_min_level_colour', 'yellow')
      call pcont


C Import the wind data
      call psetc('grib_input_file_name', '../uv200.grb')
      call pgrib

C Define the visualisation of the wind
      call pseti('wind_arrow_thickness', 1)
      call psetr('wind_thinning_factor', 3.000000)
      call psetc('wind_arrow_colour', 'gold')
      call psetc('wind_field_type', 'arrows')
      call pwind

C Import the Msl data
      call psetc('grib_input_file_name', '../msl.grb')
      call pgrib


C Define the visualisation of the msl
      call psetc('contour_level_selection_type', 'interval')
      call psetc('contour_shade', 'off')
      call psetc('legend', 'off')
      call psetc('contour_line_colour', 'black')
      call psetc('contour_lo_colour', 'black')
      call pseti('contour_highlight_thickness', 2)
      call psetc('contour_hi_colour', 'black')
      call psetc('contour_hilo', 'on')
      call psetr('contour_hilo_height', 0.250000)
      call pseti('contour_line_thickness', 1)
      call psetc('contour_label', 'off')
      call psetc('contour_highlight_colour', 'black')
      call psetc('contour_hilo_quality', 'high')
      call psetr('contour_interval', 5.000000)
      call pcont


      call psetc('map_label', 'off')
      call psetc('map_grid_colour', 'tan')
      call psetc('map_grid', 'on')
      call psetc('map_coastline_colour', 'tan')
      call psetc('map_coastline_land_shade', 'off')
      call psetc('map_coastline_sea_shade', 'off')
      call pcoast

C add the text
      call pset1c('text_lines', title, 2)
      call psetc('text_mode', 'positional')
      call psetr('text_box_y_length', 2.500000)
      call psetc('text_justification', 'left')
      call psetc('text_border', 'on')
      call psetr('text_box_x_length', 20.000000)
      call psetr('text_box_x_position', 1.500000)
      call psetc('text_html', 'true')
      call psetc('text_box_blanking', 'on')
      call psetc('text_border_colour', 'black')
      call psetr('text_box_y_position', 12.000000)
      call psetc('text_colour', 'black')
      call ptext

C Add a legend
      call psetr('legend_box_x_position', 27.000000)
      call psetc('legend_display_type', 'continuous')
      call psetc('legend_box_blanking', 'on')
      call psetr('legend_box_y_position', 0.750000)
      call psetr('legend_box_x_length', 2.000000)
      call psetc('legend_border', 'on')
      call psetc('legend_border_colour', 'black')
      call psetc('legend_box_mode', 'positional')
      call psetc('legend_text_colour', 'black')
      call psetr('legend_box_y_length', 12.000000)
      call psetc('legend', 'on')


      call pclose

      end

File: docs/tutorial/magics/msl_uv200/solution/msl_uv200.py
# (C) Copyright 1996-2016 ECMWF.
# 
# This software is licensed under the terms of the Apache Licence Version 2.0
# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0. 
# In applying this licence, ECMWF does not waive the privileges and immunities 
# granted to it by virtue of its status as an intergovernmental organisation nor
# does it submit to any jurisdiction.


#importing Magics module
from Magics.macro import *

#Setting of the output file name
output = output({"output_formats":['ps', 'png'], 
			'output_name':'msl_uv_australia'})

#Setting the coordinates of the geographical area
australia = mmap({
				    "subpage_upper_right_longitude": 190.,
                    "subpage_upper_right_latitude": -5.,
                    "subpage_lower_left_longitude": 80.,
                    "subpage_map_projection": "cylindrical",
                    "subpage_lower_left_latitude": -55.})


#Background Coastlines 
background = mcoast( {"map_coastline_sea_shade_colour": "white",
                     "map_coastline_land_shade_colour": "cream",
                     "map_grid": "off",
                     "map_coastline_land_shade": "on",
                     "map_coastline_sea_shade": "on",
                     "map_label": "off",
                     "map_coastline_colour": "tan"})

#Foreground Coastlines
foreground = mcoast( { "map_grid": "on",
                 "map_grid_colour" : "tan",
                 "map_label": "off",
                 "map_coastline_colour": "tan"})

#Import the z500 data 
msl =  mgrib({ "grib_input_file_name" : "../msl.grb",
                "grib_id": "z500"})

#Define the simple contouring for msl
msl_contour = mcont({
				"legend":"off", 
				"contour_level_selection_type": "interval",
                "contour_interval": 5.,
                "contour_line_colour": "black",
                "contour_line_thickness": 1,
                "contour_hilo": "on",
                "contour_hilo_quality": "high",
                "contour_hi_colour": "black",
                "contour_lo_colour": "black",
                "contour_hilo_height": 0.25,
                "contour_label": "off",
                "contour_highlight_colour": "black",
                "contour_highlight_thickness": 2,
                })


#Import the  wind speed at 200hPa speed200 
speed200 =  mgrib({ "grib_input_file_name" : "../speed200.grb",
                "grib_id": "t850"})


#Define the shading for the wind speed
speed200_contour = mcont({
    "legend": "on",
    "contour_level_selection_type": "level_list", 
    "contour_level_list": [30., 40., 50., 60., 70., 80., 90., 100.], 
    "contour_shade": "on", 
    "contour_shade_max_level_colour": "evergreen", 
    "contour_shade_min_level_colour": "yellow",
    "contour_shade_method": "area_fill", 
    "contour_reference_level": 0., 
    "contour_highlight": "off", 
    "contour_hilo": "on", 
    "contour_hilo_format": "(F3.0)", 
    "contour_hilo_height": 0.2, 
    "contour_hilo_type": "number", 
    "contour_hilo_suppress_radius": 30., 
    "contour_hi_min_value": 15., 
    "contour_label": "off" 
         })


#Import the  wind  at 200hPa uv200 
uv200 =  mgrib({ "grib_input_file_name" : "../uv200.grb",
                "grib_id": "uv200"})

uv200_wind = mwind({
                "legend": "on",
                "wind_field_type": "arrows",
                "wind_arrow_thickness": 1,
                "wind_thinning_factor": 3.,
                "wind_arrow_colour": "gold" })

lines =["My first line", "My second line"]

title = ptext({
           "text_lines" : lines,
           "text_html" : "true",
           "text_colour" : "black",
           "text_mode": "positional",
           "text_box_x_position": 1.5,
           "text_box_y_position": 12.,
           "text_box_x_length": 20.,
           "text_box_y_length": 2.5,
           "text_border": "on",
           "text_border_colour": "black",
           "text_box_blanking": "on",
           "text_justification" : "left"})

#add a legend
legend = mlegend({"legend":"on",
           "legend_text_colour":"black",
           "legend_box_mode": "positional",
           "legend_box_x_position": 27.,
           "legend_box_y_position": 0.75,
           "legend_box_x_length": 2.,
           "legend_box_y_length": 12.,
           "legend_border": "on",
           "legend_border_colour": "black",
           "legend_box_blanking": "on",
           "legend_display_type": "continuous" })

#To the plot
plot(output, australia, background, 
		speed200, speed200_contour,
		uv200, uv200_wind,
		msl, msl_contour, 
		foreground, title, legend)
